yt_dlp/extractor/sohu.py

```diff
@@ -132,7 +157,7 @@ def _fetch_data(vid_id, mytv=False):
                 allot = format_data['allot']
 
                 data = format_data['data']
-                clips_url = data['clipsURL']
+                clip_url = traverse_obj(data, (('clipsURL', 'mp4PlayUrl'), i, {url_or_none}), get_all=False)
```

Comment:
Actually, I wonder if an `ExtractorError` would be better.

This for loop iterates over different parts of the multi-part list.
I think it is better to throw an error than silently ignore errors, because we expect/require a valid `clip_url` here to properly download full video. A program throws no warning or error and gives an empty playlist or a partial video with missing parts is more confusing than a program that just throws error.

What's your idea on this?

Reply:
```suggestion
                clip_url = traverse_obj(data, (('clipsURL', 'mp4PlayUrl'), i, {url_or_none}), get_all=False)
                if not clip_url:
                    raise ExtractorError(f'Failed to extract clip_url from format_data, part index={i}')
```

Reply:
Good point, agreed. We could simplify the message a bit, I think
```suggestion
                clip_url = traverse_obj(data, (('clipsURL', 'mp4PlayUrl'), i, {url_or_none}), get_all=False)
                if not clip_url:
                    raise ExtractorError(f'Unable to extract url for clip {i}')
```